tests/common.py

```diff
@@ -263,10 +265,18 @@ def assertMetric(self, metric_name, value=None, tags=None, count=None, at_least=
                     continue
                 if tags is not None and sorted(tags) != sorted(mdata.get("tags", [])):
                     continue
+                if hostname is not None and mdata['hostname'] != hostname:
+                    continue
 
                 candidates.append((m_name, ts, val, mdata))
 
-        self._candidates_size_assert(candidates, count=count, at_least=at_least)
+        try:
+            self._candidates_size_assert(candidates, count=count, at_least=at_least)
+        except AssertionError:
+            log.error("No candidates for {0} (value: {1}, tags: {2}, count: {3}, at_least: {4}, hostname: {5})"\
+                .format(metric_name, value, tags, count, at_least, hostname))
+            raise
+
```

Comment:
I didn't find the debug log visible enough when testing, especially when you have a lot of metrics. This log line made my debugging easier. I can change the message though.